integration/apps/hpl_cpu/hpl_cpu.py

```diff
@@ -0,0 +1,155 @@
+        self.P = process_grid_ratios[num_nodes]['P']
+        self.Q = process_grid_ratios[num_nodes]['Q']
+
+        self.N = int(round(math.sqrt(dram_for_app / 8)))
+        self._cpu_per_rank = cores_per_node
+
+        print(f'DRAM reserved for APP: {dram_for_app} Cores for app: {cores_per_node} N={self.N}')
+
+    def get_bash_setup_commands(self):
+        input_file = textwrap.dedent(f'''
+        HPLinpack benchmark input file
+        Innovative Computing Laboratory, University of Tennessee
+        HPL.out      output file name (if any)
+        6            device out (6=stdout,7=stderr,file)
+        1            # of problems sizes (N)
+        {self.N}     Ns
+        1            # of NBs
+        {self.NBs}   NBs
+        0            PMAP process mapping (0=Row-,1=Column-major)
+        1            # of process grids (P x Q)
+        {self.P}     Ps
+        {self.Q}     Qs
+        16.0         threshold
+        1            # of panel fact
+        1            PFACTs (0=left, 1=Crout, 2=Right)1
+        1            # of recursive stopping criterium
+        4            NBMINs (>= 1)
+        1            # of panels in recursion
+        2            NDIVs
+        1            # of recursive panel fact.
+        1            RFACTs (0=left, 1=Crout, 2=Right)
+        1            # of broadcast
+        0            BCASTs (0=1rg,1=1rM,2=2rg,3=2rM,4=Lng,5=LnM)
+        1            # of lookahead depth
+        0            DEPTHs (>=0)
+        2            SWAP (0=bin-exch,1=long,2=mix)
+        64           swapping threshold
+        0            L1 in (0=transposed,1=no-transposed) form
+        0            U  in (0=transposed,1=no-transposed) form
+        1            Equilibration (0=no,1=yes)
+        8            memory alignment in double (> 0)
+        EOF
+        ''')
+        # KMP_HW_SUBSET: forces 1 thread per core to be used. Technically this
+        # should not be needed since --geopm-hyperthreads-disable is set below.
+        # OMP_NUM_THREADS: # number of cores per node. Technically this should
+        # not be needed since setting _cpu_per_rank will prompt the launcher to
+        # choose that many cores.
+        # MKL_NUM_THREADS: # matches with above, effective if linked against MKL.
+        env_file = textwrap.dedent(f'''
+        export KMP_HW_SUBSET=1t
+        export OMP_NUM_THREADS={self._cpu_per_rank}
+        export MKL_NUM_THREADS={self._cpu_per_rank}
+        EOF
+        ''')
+
+        return f'cat > hpl_env.sh << EOF {env_file} source ./hpl_env.sh; cat > ./HPL.dat << EOF {input_file}'
+
+    def get_rank_per_node(self):
+        return 1
+
+    def get_cpu_per_rank(self):
+        return self._cpu_per_rank
+
+    def get_bash_exec_path(self):
+        return self.exe_path
+
+    def get_bash_exec_args(self):
+        return ''
+
+    def get_custom_geopm_args(self):
+        return ['--geopm-ctl=application',
```

Comment:
I recall that this is required for some reason and process mode does not work.  Please capture this here and/or in the README so we do not lose this detail.

Reply:
I put a bunch of info in the README.md.

Reply:
...and comments in code.